omod/src/main/java/org/openmrs/module/referenceapplication/filter/RequireLoginLocationFilter.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.referenceapplication.filter;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.referenceapplication.ReferenceApplicationConstants;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.WebConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Intercepts authenticated requests to check if the logged in user has selected a location and will
+ * redirect them to the login page so they can select one in case they haven't yet.
+ */
+public class RequireLoginLocationFilter implements Filter {
+	
+	private static final Logger logger = LoggerFactory.getLogger(RequireLoginLocationFilter.class);
+	
+	private static final String[] STATIC_CONTENT_EXTENSIONS = new String[] { "js", "css", "ico", "png", "gif", "jpg",
```

Comment:
What happens for a static content with an extension that is not in this list?

Reply:
Doesn't get pulled in until you are logged in worse case scenario.
It isn't a big deal since this will only affect the login and home page which actually have minimal content and functionality.

Reply:
I can change this to allow all requests and only apply the filter to **.page** requests that way we don't have to deal with static content or other extensions. I will apply the change